Container.py

```diff
@@ -167,35 +182,35 @@ def add(output_register: GeneralRegister, input_register1: GeneralRegister, inpu
     if not progressive_beams:
         return []
     cstate_manager_global = GAMES[game_version]["cstate_manager_global"]
-    charge_beam_offset = 0x7F
+    charge_beam_offset = 0x7C
     instructions: List = [
         # Step 0: Get the player state address
         *_load_player_state_to_r6(game_version),
 
         # Step 1: Get the current beam from the player state
-        lbz(r5, 0xB, r6),  # Load the current beam value
+        lwz(r5, 0x8, r6),          # Load the current beam value
 
-        # Step 2: Read the value at the progressive beam address
-        lis(r7, GAMES[game_version]["PROGRESSIVE_BEAM_ADDRESS"] >> 16),  # Load upper 16 bits of progressive beam address
-        ori(r7, r7, GAMES[game_version]["PROGRESSIVE_BEAM_ADDRESS"] & 0xFFFF),  # Load lower 16 bits of progressive beam address
-        add(r7, r7, r5),  # Add 0 to the address (no offset
-        lbz(r8, 0, r7),  # Load the value at the progressive beam address
+        # Step 2: Read the value at the beam address in CPlayerState::PowerUps[41]

```

Comment:
Can you give just a general description of what's going on here? I see the comments so I have a vague idea of the 'what', but I'm not completely following the 'why' and I definitely want to make sure I understand this well enough to maintain it solo if necessary before merging it in 😂

Reply:
Reading at CPlayerState::PowerUps[beam].x4_capacity.

Where 0 is no beam
Where 1 is beam
Where 2 is charge beam